cachito/workers/pkg_managers/rubygems.py

```diff
@@ -478,3 +479,18 @@ def _get_metadata(package_root, request):
     repo_name = get_repo_name(request["repo"]).split("/")[-1]
 
     return repo_name + relative_path, request["ref"]
+
+
+def get_rubygems_hosted_url_with_credentials(username, password, request_id):
+    """
+    Get URL of a RubyGems hosted repo for the request with hardcoded username and a password
+
+    :param username: the username that has read access on the RubyGems hosted repo
+    :param password: password for the user`
+    :param int request_id: the ID of the request this repository is for
+    :return:
+    """
+    config = get_worker_config()
+    repo_name = get_rubygems_hosted_repo_name(request_id)
+    return f"http://{username}:{password}@localhost:8082/repository/{repo_name}/"
```

Comment:
@chmeliik @lkolacek @ejegrova  This URL is used in `.bundle/config` file to redirect Bundler to use it as a proxy repo for downloading RubyGems. I would love to get the address from some config, but I don't know what to use. There is `cachito_nexus_rubygems_proxy_url = "http://nexus:8081/repository/cachito-rubygems-proxy/"` in the worker config, but it won't work, because docker compose remaps Nexus' port 8081 to 8082 (at least in development mode). Do you have any idea what I could do about it?

Reply:
You could do a similar thing that pip does for the `PIP_INDEX_URL`

(TL;DR - use `config.cachito_nexus_url`)

https://github.com/containerbuildsystem/cachito/blob/07f5cb1ea8307b08ac8a3b1746c28d03fda5c72c/cachito/workers/tasks/pip.py#L112-L114

https://github.com/containerbuildsystem/cachito/blob/07f5cb1ea8307b08ac8a3b1746c28d03fda5c72c/cachito/workers/pkg_managers/pip.py#L1849-L1859

Reply:
The reason I was asking is that using `config.cachito_nexus_url` won't work with my development setup (but I don't know how Cachito is deployed internally, so it might not have the same problem), because bundler won't be able to download any dependencies from that URL (http://nexus:8081). Anyways, I implemented it this way, but in order for me to work locally, I'd probably have to change `config.cachito_nexus_url` to `http://localhost:8082`, so I'm still a bit confused.

Reply:
Huh, yeah you're right. Now that you mention it, I think pip has the same problem, in order to install from Nexus we would have to `sed 's/nexus:8081/localhost:8082/'` in our testing.

I think the problem is that `cachito_nexus_url` has to be accessible from inside the docker-compose environment, where `localhost:8082` doesn't work. The real deployment does not have this issue because the URL is accessible both internally and externally.

To fix that perhaps we could have one internal and one external nexus url in the config, but we never bothered with it

Reply:
Oh, in that case, I'll keep it as it is for now